transplant/transplant_remote.m

```diff
@@ -110,14 +110,29 @@ function transplant_remote(msgformat, url, zmqname, is_zombie)
                     if isKey(msg, 'nargout') && msg('nargout') >= 0
                         resultsize = msg('nargout');
                     else
+                        % nargout(fun) throws fails if fun is a method
+                        % nargout(msg('name')) can work on class methods, but
+                        % no way to specify class name in MATLAB x-(
+                        % do extra check on class name using which(msg('name'))
                         try
-                            resultsize = nargout(fun);
-                        catch % nargout fails if fun is a method:
-                            try
-                                resultsize = nargout(msg('name'));
-                            catch
-                                resultsize = -1;
+                            [~, funType] = which(msg('name'));
+                            funType = strsplit(funType,' ');
+                            if numel(funType)==2 && strcmp(funType{2},'method')
+                                argTemp = msg('args');
+                                className = funType{1};
+                                if ~isempty(argTemp) && isa(argTemp{1},className)
+                                    % the function name corresponds to the
+                                    % class of the first argument
+                                    resultsize = nargout(msg('name'));
+                                end
                             end
+                            
+                            if isempty(resultsize)
```

Comment:
if `fun` is a function, `resultsize` will not exist, and this will throw an error. (right?)